### PR TITLE
Add a GitHub Actions build

### PR DESCRIPTION
**Pull request by mbautin**
Adding a basic build of non-test code (what's needed by the release package) in a CentOS Docker image on GitHub Actions.

### CODE REVIEW

File: python/yb/download_and_extract_archive.py [hunk 1]
```diff
@@ -191,7 +191,6 @@ def download_and_extract(url, dest_dir_parent, local_cache_dir, nfs_cache_dir):
             logging.info("Failed creating directory '%s': %s", local_cache_dir, ex)
 
     check_dir_exists_and_is_writable(local_cache_dir, "Local cache")
-    check_dir_exists_and_is_writable(nfs_cache_dir, "NFS cache")
```
Comment (rajukumaryb) [inline python/yb/download_and_extract_archive.py:194]: With this change, `./yb_build.sh tsan --clean --dltp` was able to make progress. Thanks!

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  # This workflow contains a single job called "build"
+  build:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        name:
+          #- macOS
+          - "CentOS 7"
+          #- "Ubuntu 18.04"
```
Comment (ttyusupov) [inline .github/workflows/build.yml:29]: Do we need to keep commented-out OS names in a final commit?

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    branches:
+      - master
+    paths-ignore:
+      - README.md
+
+  pull_request:
+    branches:
+      - master
+    paths-ignore:
+      - README.md
```
Comment (ttyusupov) [inline .github/workflows/build.yml:16]: Do we need to ignore changes to docs, managed directories and might be something else?

Comment (mbautin) [reply .github/workflows/build.yml:16]: Added the ignore item for docs. Will add more items in a follow-up pull request once this pull request lands.

**Pull request merged**